src/libraries/Common/src/Interop/Unix/System.Native/Interop.ForkAndExecProcess.cs

```diff
@@ -49,20 +49,17 @@ internal static partial class Sys
 
         private static unsafe void AllocNullTerminatedArray(string[] arr, ref byte** arrPtr)
         {
-            int arrLength = arr.Length + 1; // +1 is for null termination
+            int arrLength = checked(arr.Length + 1); // +1 is for null termination
```

Comment:
This can't actually overflow, can it? array.Length <= Array.MaxLength < int.MaxValue - 1

Reply:
Mono isn't bound by this same constraint. It also seems fragile to depend on the runtime never allowing this value to increase in the future.